openquake/engine/postproc/compute_mrd.py

```diff
@@ -0,0 +1,106 @@
+    """
+    :param ctx: a context array
+    :param N: the total number of sites
+    :param cmaker: a ContextMaker instance
+    :param crosscorr: cross correlation model
+    :param str imt1: the first Intensity Measure Type
+    :param str imt1: the second Intensity Measure Type
+    :param meabins: bins for the means
+    :param sigbins: bins for the sigmas
+    :param method: a string 'direct' or 'indirect'
+    :returns: 4D-matrix with shape (L1, L1, N, G)
+    """
+    mrd = calc_mean_rate_dist(ctx, N, cmaker, crosscorr,
+                              imt1, imt2, meabins, sigbins, method)
+    return {g: mrd[:, :, :, i] for i, g in enumerate(cmaker.gidx)}
+
+
+def combine_mrds(acc, rlzs_by_g, rlz_weight):
+    g = next(iter(acc))  # first key
+    out = numpy.zeros(acc[g].shape)  # shape (L1, L1, N)
+    for g in acc:
+        value = acc[g]
+        for rlz in rlzs_by_g[g]:
+            out += rlz_weight[rlz] * value
+    return out
+
+
+def main(parent_id, config):
+    """
+    :param parent_id: filename or ID of the parent calculation
+    :param config: name of a .toml file with parameters imt1, imt2, crosscorr
+
+    NB: this is meant to work only for parametric ruptures!
+    """
+    try:
+        parent_id = int(parent_id)
+    except ValueError:
+        # passing a filename is okay
+        pass
+    parent = datastore.read(parent_id)
+    dstore, log = datastore.build_dstore_log(parent=parent)
+    with open(config) as f:
+        dic = toml.load(f)
+    imt1 = dic['imt1']
+    imt2 = dic['imt2']
+    method = dic.get('method', 'indirect')
+    crosscorr = getattr(cross_correlation, dic['cross_correlation'])()
+    meabins = dic['meabins']
+    sigbins = dic['sigbins']
+    with dstore, log:
+        oq = parent['oqparam']
+        N = len(parent['sitecol'])
+        L1 = oq.imtls.size // len(oq.imtls) - 1
+        if L1 > 24:
+            logging.warning('There are many intensity levels (%d), the '
+                            'calculation can be pretty slow', L1 + 1)
+        assert N <= 10, 'Too many sites: %d' % N
+        cmakers = contexts.read_cmakers(parent)
+        ctx_by_grp = contexts.read_ctx_by_grp(dstore)
+        n = sum(len(ctx) for ctx in ctx_by_grp.values())
+        logging.info('Read {:_d} contexts'.format(n))
+        blocksize = numpy.ceil(n / oq.concurrent_tasks)
+        dstore.swmr_on()
+        smap = parallel.Starmap(compute_mrd, h5=dstore)
+        for grp_id, ctx in ctx_by_grp.items():
+            cmaker = cmakers[grp_id]
+            for slc in general.gen_slices(0, len(ctx), blocksize):
+                smap.submit((ctx[slc], N, cmaker, crosscorr, imt1, imt2,
+                             meabins, sigbins, method))
+        acc = smap.reduce()
+        mrd = dstore.create_dset('_mrd', float, (L1, L1, N))
+        mrd[:] = combine_mrds(acc, dstore['rlzs_by_g'][:], dstore['weights'][:])
+        return mrd[:]
+
+
+if __name__ == '__main__':
+    sap.run(main)
```

Comment:
There is a big issue here, @mmpagani . The file openquake/engine/postproc/compute_mrd.py must not be added here, because we have already a file openquake/calculators/postproc/compute_mrd.py that does its job. I guess you started from an old branch, the location of the postproc directory was changed many months ago.

Reply:
Shoot. I will move it to the right place. Grazie Michele